### PR TITLE
[CI] [GHA] Introduce nGraph Python Unittests

### PR DESCRIPTION
**Pull request by akashchi**
### Details:
This PR introduces the nGraph Python Unittests from the [Linux nGraph ONNX Azure pipeline](https://github.com/openvinotoolkit/openvino/blob/master/.ci/azure/linux_ngraph_onnx.yml). 

**_TL;DR_**: this PR makes the [Linux nGraph ONNX Azure pipeline](https://github.com/openvinotoolkit/openvino/blob/master/.ci/azure/linux_ngraph_onnx.yml) obsolete.

There are two types of tests in the Linux nGraph ONNX Azure pipeline: **Release** and **Debug**.
* **Release**: uses `tox && tox -e zoo_models` to execute tests. 
* **Debug**: uses `tox` to execute tests.

Both of them use settings from this [setup.cfg](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/setup.cfg):
* **Release**: `pytest --backend={env:OV_BACKEND} /openvino/src/frontends/onnx/tests/tests_python/test_zoo_models.py -v -n 4 --forked -k 'not _cuda' --model_zoo_xfail` - it is said to be obsolete as it requires models from a not-available-anymore share: https://onnxruntimetestdata.blob.core.windows.net/models/20191107.zip; see ticket 58676. It is working in Azure as the machines there use a local copy of the models from the said share.
* **Debug**: 
  * `pytest tests -m "not template_extension" -v -k 'not _cuda' --ignore=tests/test_utils` - already covered by other steps in the `Python Unittests` Job.
  * `pytest --backend={env:OV_BACKEND} tests_compatibility/test_ngraph -v -k 'not _cuda' --ignore=tests_compatibility/test_onnx/test_zoo_models.py` - newly added via this PR.
  * `pytest --backend={env:OV_BACKEND} /openvino/src/frontends/onnx/tests -v --ignore=/openvino/src/frontends/onnx/tests/tests_python/test_zoo_models.py` - already present in the `Python Unittests` Job, in the `Python ONNX operators tests` step.

**Release** configuration is obsolete, and **Debug** configuration is partly present, once this PR is ~~hopefully~~ merged, the Linux GHA pipeline will fully incorporate the Azure nGraph ONNX pipeline.

### Tickets:
 - *122737*

### CODE REVIEW

**Comment by akashchi** [comment]
The newly added stage is passing:
* Linux: https://github.com/openvinotoolkit/openvino/actions/runs/6546459665/job/17777362014?pr=20514
* Win: https://github.com/openvinotoolkit/openvino/actions/runs/6546459662/job/17778153423?pr=20514
* Mac: https://github.com/openvinotoolkit/openvino/actions/runs/6546459632/job/17777244615?pr=20514

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -1008,11 +1008,16 @@ jobs:
...
-          # Skip test_onnx/test_zoo_models and test_onnx/test_backend due to long execution time - ONNX Model Zoo tests are run separately
+          # Skip test_onnx/test_zoo_models due to long execution time - ONNX Model Zoo tests are run separately
           python3 -m pytest -sv ${OPENVINO_REPO}/src/frontends/onnx/tests -k 'not cuda' \
             --junitxml=${INSTALL_TEST_DIR}/TEST-onnx_frontend.xml \
             --ignore=${OPENVINO_REPO}/src/frontends/onnx/tests/test_python/test_zoo_models.py
 
+      - name: nGraph Tests
+        run: |
+          python3 -m pytest -v ${OPENVINO_REPO}/src/bindings/python/tests_compatibility/test_ngraph -k 'not _cuda' \
+            --junitxml=${INSTALL_TEST_DIR}/TEST-ngraph.xml
```
Comment (ilya-lavrenov) [inline .github/workflows/linux.yml:1019]: how is it different from Python API 1.0 tests?

Comment (akashchi) [reply .github/workflows/linux.yml:1019]: Seems like it is no different 🥲. So I suppose the Linux nGraph ONNX pipeline could be deleted if there are no more missing parts from there in the Linux GHA pipeline.

Comment (ilya-lavrenov) [reply .github/workflows/linux.yml:1019]: Linux ONNX tests on ONNX model zoo, which we don't have in GHA I suppose

Comment (akashchi) [reply .github/workflows/linux.yml:1019]: But we cannot have them as:
image

**Pull request closed without merging**